### ISSUE TITLE
MA0158 is (now) duplicated by IDE0330

### ISSUE THREAD

**Issue by viceroypenguin** [meziantou/Meziantou.Analyzer#760]
- Rule: MA0158
- Link to the equivalent rule documentation: [IDE0330](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0330) (link not valid yet)

evidence: 
![image](https://github.com/user-attachments/assets/9256af56-06dc-45f6-93cb-ff82f50a56ba)

**Comment by meziantou** [comment]
I've added IDE0330 to https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/comparison-with-other-analyzers.md

Note: While there are similar, MA0153 seems to report more cases:

```c#
public class Sample
{
    private readonly object obj = new(); // MA153 and IDE0330
    public void B()
    {
        lock (obj)
        {
        }
    }

    public void A()
    {
        object a = new(); // MA153
        lock (a)
        {
        }
    }
}
```
